src/pybind/mgr/volumes/fs/purge_queue.py

```diff
@@ -110,4 +110,10 @@ def get_next_job(self, volname, running_jobs):
         return get_trash_entry_for_volume(self.fs_client, self.vc.volspec, volname, running_jobs)
 
     def execute_job(self, volname, job, should_cancel):
-        purge_trash_entry_for_volume(self.fs_client, self.vc.volspec, volname, job, should_cancel)
+        if not self.vc.mgr.purge_trash:
```

Comment:
I think the problem here is that the `job` is already added to `self.jobs` (i.e., running_jobs). So if the subvolume is deleted when purge_trash is `False`, the subvolume would never be purged even after `purge_trash` is set `True`  because the `get_next_job` would skip fetching already running_jobs ?